oneclick.py

```diff
@@ -91,38 +110,26 @@ def create_identifier_strings(cls, identifiers):
 
     @classmethod
     def from_config(cls, _db):
-        config = Configuration.integration(Configuration.ONECLICK_INTEGRATION, required=True)
-        property_names = [
-            'library_id',
-            'username',
-            'password',
-            'remote_stage', 
-            'url', 
-            'basic_token', 
-            'ebook_loan_length', 
-            'eaudio_loan_length'
-        ]
-        property_values = {}
-        for name in property_names:
-            value = config.get(name)
-            if value:
-                value = value.encode("utf8")
-            property_values[name] = value
-
-        if len(property_values.values()) == 0:
-            cls.log.info("No OneClick client configured.")
-            raise ValueError("No OneClick client configured.")
-
-
-        api = cls(_db, library_id=property_values['library_id'], 
-            username=property_values['username'], password=property_values['password'], 
-            remote_stage=property_values['remote_stage'], base_url=property_values['url'], 
-            basic_token=property_values['basic_token'], 
-            ebook_loan_length=property_values['ebook_loan_length'], 
-            eaudio_loan_length=property_values['eaudio_loan_length'])
-
-        return api
+        """Load a OneClickAPI instance for the 'default' OneClick
+        collection.
+        """
+        library = Library.instance(_db)
+        collections = [x for x in library.collections
+                       if x.protocol == Collection.ONE_CLICK]
+        if len(collections) == 0:
+            # There are no OneClick collections configured.
+            return None
 
+        if len(collections) > 1:
+            raise ValueError(
+                "Multiple OneClick collections found for one library. This is not yet supported."
+            )
+        [collection] = collections 
+
+        try:
+            return cls(_db, collection)
+        except CannotLoadConfiguration, e:
```

Comment:
After looking at the code I don't think it's necessary to muffle these errors. If a Collection is present but improperly configured, an exception _should_ be raised here.